Add exhaustive eager boxing test

clackhan opened the pull request:
添加详尽的eager boxing测试用例（覆盖了所有的常规情况，以及S(2)、S(4)这些特殊情况用例），目前已知的不支持的eager boxing 类型为非均匀切分的2d boxing，这种情况将在 [Oneflow-Inc/oneflow/pull/7768](https://github.com/Oneflow-Inc/oneflow/pull/7768) 中解决

wyg1997 commented on python/oneflow/test/modules/test_eager_boxing_exhaustive.py:
```diff
@@ -0,0 +1,250 @@
+
+import numpy as np
+import oneflow as flow
+import os
```
这个导包应该在最上面，和 unittest、collections 放一起

clackhan replied:
以放到上面

wyg1997 commented on python/oneflow/test/modules/test_eager_boxing_exhaustive.py:
```diff
@@ -0,0 +1,250 @@
+def _test_eager_boxing_normal_1d_exhaustive_testing(
+    test_case, shape, in_device, out_device, in_device_list, out_device_list
+):
+    import itertools
```
这个要不也放在最上面一起导了？

clackhan replied:
已修改

wyg1997 commented:
可以贴一下单测 print 出来的结果

clackhan commented:
> 可以贴一下单测 print 出来的结果

总共有 ```3 * 2 * 2 * 3 * 2 + 2 * 2 + 2 * 2 * 3 * 2 + 2 * 2 = 104```种输出😂：

```
(oneflow) [hanbinbin@oneflow-16 modules]$ ONEFLOW_TEST_DEVICE_NUM=4 python3 -m oneflow.distributed.launch --nproc_per_node 4 test_eager_boxing_exhaustive.py --failfast --verbose
test_eager_boxing_1d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... Environment has been initialized, this env init will do nothing.
test_eager_boxing_1d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... Environment has been initialized, this env init will do nothing.
test_eager_boxing_1d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... Environment has been initialized, this env init will do nothing.
test_eager_boxing_1d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... Environment has been initialized, this env init will do nothing.
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [1, 2, 3], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cpu', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [1, 2, 3], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cpu', 'cuda', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [1, 2, 3], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cpu', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [1, 2, 3], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=2), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=4), oneflow.sbp.partial_sum), (16, 16, 5, 8, 7, 6), 'cuda', 'cuda', [0, 1, 2, 3], [0, 1, 3])]
ok
test_eager_boxing_2d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... 0 unsuported boxing 1d type
[]
ok
ok
test_eager_boxing_2d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... test_eager_boxing_2d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... ok
test_eager_boxing_2d_special_split_axis (__main__.TestEagerBoxingSpecialSplitAxisExhaustiveTesting) ... 2 unsuported boxing 2d type
[(((oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4)), (oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2))), 'cpu', 'cpu'), (((oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2)), (oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4))), 'cpu', 'cpu')]
2 unsuported boxing 2d type
[(((oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4)), (oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2))), 'cpu', 'cuda'), (((oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2)), (oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4))), 'cpu', 'cuda')]
2 unsuported boxing 2d type
[(((oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4)), (oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2))), 'cuda', 'cpu'), (((oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2)), (oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4))), 'cuda', 'cpu')]
2 unsuported boxing 2d type
[(((oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4)), (oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2))), 'cuda', 'cuda'), (((oneflow.sbp.split(axis=4), oneflow.sbp.split(axis=2)), (oneflow.sbp.split(axis=2), oneflow.sbp.split(axis=4))), 'cuda', 'cuda')]
ok
ok
test_eager_boxing_normal_1d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... ok
test_eager_boxing_normal_1d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... ok
test_eager_boxing_normal_1d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... test_eager_boxing_normal_1d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... 0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (12, 12), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (12, 12), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (12, 12), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (12, 12), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (12, 12), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (12, 12), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (18, 24), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cpu', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cpu', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (18, 24), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cuda', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cpu', 'cuda', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (18, 24), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cpu', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cpu', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
0 unsuported boxing 1d type
[]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (18, 24), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cuda', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (18, 24), 'cuda', 'cuda', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (15, 17), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [1, 2, 3], [0, 1, 3])]
1 unsuported boxing 1d type
[((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cpu', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (15, 17), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [1, 2, 3], [0, 1, 3])]
1 unsuported boxing 1d type
[((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cpu', 'cuda', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (15, 17), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [1, 2, 3], [0, 1, 3])]
1 unsuported boxing 1d type
[((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cpu', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
[]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [0, 1], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [0, 1], [0, 1, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [0, 1], [0, 1, 2, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [0, 1], [0, 1, 2, 3])]
5 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.broadcast), (15, 17), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.broadcast, oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3]), ((oneflow.sbp.partial_sum, oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [1, 2, 3], [0, 1, 3])]
1 unsuported boxing 1d type
[((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [1, 2, 3], [0, 1, 2, 3])]
2 unsuported boxing 1d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [0, 1, 2, 3], [0, 1, 3]), ((oneflow.sbp.split(axis=1), oneflow.sbp.partial_sum), (15, 17), 'cuda', 'cuda', [0, 1, 2, 3], [0, 1, 3])]
0 unsuported boxing 1d type
ok
ok
[]
test_eager_boxing_symmetric_2d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... test_eager_boxing_symmetric_2d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... ok
ok
test_eager_boxing_symmetric_2d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... test_eager_boxing_symmetric_2d_exhaustive_testing (__main__.TestEagerBoxingSymmetricExhaustiveTesting) ... 2 unsuported boxing 2d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)), (oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0))), ((oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0)), (oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)))]
2 unsuported boxing 2d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)), (oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0))), ((oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0)), (oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)))]
2 unsuported boxing 2d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)), (oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0))), ((oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0)), (oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)))]
2 unsuported boxing 2d type
[((oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)), (oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0))), ((oneflow.sbp.split(axis=1), oneflow.sbp.split(axis=0)), (oneflow.sbp.split(axis=0), oneflow.sbp.split(axis=1)))]
ok
ok
ok

----------------------------------------------------------------------
Ran 4 tests in 31.135s

OK


----------------------------------------------------------------------
----------------------------------------------------------------------
Ran 4 tests in 31.107s
Ran 4 tests in 31.115s


OK
OK
ok

----------------------------------------------------------------------
Ran 4 tests in 31.129s

OK
```
